packages/grid/_modules_/grid/hooks/features/rows/useGridEditRows.ts

```diff
@@ -110,7 +140,39 @@ export function useGridEditRows(
     [apiRef, forceUpdate, logger, setGridState],
   );
 
-  const getCellMode = React.useCallback(
+  const setRowMode = React.useCallback<GridEditRowApi['setRowMode']>(
+    (id, mode) => {
+      setGridState((state) => {
+        const newEditRowsState: GridEditRowsModel = { ...state.editRows };
+        if (mode === 'edit') {
+          newEditRowsState[id] = {};
+          columns.forEach((column) => {
+            const cellParams = apiRef.current.getCellParams(id, column.field);
+            if (cellParams.isEditable) {
+              newEditRowsState[id][column.field] = { value: cellParams.value };
+            }
+          });
+        } else {
+          delete newEditRowsState[id];
+        }
+        return { ...state, editRows: newEditRowsState };
+      });
+      forceUpdate();
+    },
+    [apiRef, columns, forceUpdate, setGridState],
+  );
+
+  const getRowMode = React.useCallback<GridEditRowApi['getRowMode']>(
+    (id) => {
+      if (props.editMode === 'cell') {
+        return 'view';
```

Comment:
I think we should move those into an `enum`. Typing it every time seems error-prone.